templates/jupyterhub_config.py.j2

```diff
@@ -22,12 +22,47 @@ c.PAMAuthenticator.open_sessions = False
 # Listen on all interfaces, since hub should be reachable from spawned nodes
 c.JupyterHub.hub_ip = '0.0.0.0'
 
+# Assign Qhub Spawner
 class QHubHPCSpawner(SlurmSpawner):
     req_conda_environment_prefix = Unicode('',
         help="Conda environment prefix to launch jupyterlab"
     ).tag(config=True)
 
+c.JupyterHub.template_paths = []
+c.JupyterHub.extra_handlers = []
+
+{% if cdsdashboards.enabled %}
+from cdsdashboards.app import CDS_TEMPLATE_PATHS
+from cdsdashboards.hubextension import cds_extra_handlers
+from cdsdashboards.hubextension.spawners.variablemixin import VariableMixin, MetaVariableMixin
+
+c.VariableMixin.extra_presentation_launchers = {'qhub-bokeh': {
+       'cmd': ['/opt/conda/envs/jupyterlab/bin/jhsingle-native-proxy'],            
+       'args': ['--destport=0', '/opt/conda/envs/jupyterlab/bin/python',
```

Comment:
path need to be templated with jinja to get correct. e.g. "{{ miniforge.home }}/envs/{{ jupyterhub.lab_environment }}/bin/python"